src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(blocks[2][1], gridBagConstraints);
+
+		blocks[2][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(blocks[2][2], gridBagConstraints);
+	}
+
+	public void populatePanel(GameSession gameSession, boolean isRefresh) {
+
+		this.gameSession = gameSession;
+		this.selectedCell = null;
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][1].populate(gameSession.getGameBoard().getBlock(1),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][2].populate(gameSession.getGameBoard().getBlock(2),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[1][0].populate(gameSession.getGameBoard().getBlock(3),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][1].populate(gameSession.getGameBoard().getBlock(4),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][2].populate(gameSession.getGameBoard().getBlock(5),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[2][0].populate(gameSession.getGameBoard().getBlock(6),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][1].populate(gameSession.getGameBoard().getBlock(7),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][2].populate(gameSession.getGameBoard().getBlock(8),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+	}
+	
+	public void populatePencilMark(GameSession gameSession) {
+
+		this.gameSession = gameSession;
+
+		blocks[0][0].populatePencilMark(gameSession.getGameBoard().getBlock(0),
+				gameSession);
+		blocks[0][1].populatePencilMark(gameSession.getGameBoard().getBlock(1),
+				gameSession);
+		blocks[0][2].populatePencilMark(gameSession.getGameBoard().getBlock(2),
+				gameSession);
+
+		blocks[1][0].populatePencilMark(gameSession.getGameBoard().getBlock(3),
+				gameSession);
+		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
+				gameSession);
+		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
+				gameSession);
+
+		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
+				gameSession);
+		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
+				gameSession);
+		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
+				gameSession);
+	}
+
+	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
+		
+		if (null != selectedCell) {

```

Comment:
Yet another thing you will find other people with different opinions (but
not as equally popular as calling getters and using this keywords).
To give you a big more background, the convention was popular among old
timers using C where single '=' can be condition and assignment. So if you
start with null, you detect a problem early but if you put null after, you
might accidentally assign null to a variable. This, however, isn't really
possible for Java as we need double '=' for conditions. But some Sages
continued to use regardless.

Another perspective is the constant vs. variable being tested. A simple
example is Boolean objects and equals method.
When comparing a constant to a variable for a condition, with equals
method, the constant should be on the left side of equals method call.
Otherwise, we run a risk of null pointer exception.
In this context, Boolean.TRUE.equals(c) is the right way of coding, and it
is Yoda-like, too. 'TRUE, c equals.'
Following this fashion, we can place null on the left side of conditional
statement.

//==
Boolean a = Boolean.FALSE;
Boolean b = Boolean.TRUE;
Boolean c = null;

/* right way of coding */
if (Boolean.TRUE.equals(c)) {
...
}

/* null pointer risk */
if (c.equals(Boolean.TRUE)) {
...
}
//==

This, again, boils down to personal preference and style which I adopted
and have been using. I am actually surprised that you've never seen anyone
doing this.
We aren't many, but we are out there ;)









On Sun, Jul 12, 2015 at 10:53 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34430636>:
>
> > +		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
> > +				gameSession);
> > +		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
> > +				gameSession);
> > +
> > +		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
> > +				gameSession);
> > +		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
> > +				gameSession);
> > +		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
> > +				gameSession);
> > +	}
> > +
> > +	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
> > +		
> > +		if (null != selectedCell) {
>
> Weird, that is. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34430636>.
>